x-pack/plugins/security/server/usage_collector/security_usage_collector.ts

```diff
@@ -87,7 +92,8 @@ export function registerSecurityUsageCollector({ usageCollection, config, licens
         };
       }
 
-      const auditLoggingEnabled = allowAuditLogging && config.audit.enabled;
+      const auditLoggingEnabled =
+        (allowAuditLogging || allowLegacyAuditLogging) && config.audit.enabled;
```

Comment:
I think there's a small logic bug here. This would incorrectly report that audit logging is enabled if a user configured an appender (opting into the non-legacy audit logger) with a standard license.

I think something like this would do the trick:
```diff
diff --git a/x-pack/plugins/security/server/usage_collector/security_usage_collector.ts b/x-pack/plugins/security/server/usage_collector/security_usage_collector.ts
index b85cf9f293d..64a58658864 100644
--- a/x-pack/plugins/security/server/usage_collector/security_usage_collector.ts
+++ b/x-pack/plugins/security/server/usage_collector/security_usage_collector.ts
@@ -92,8 +92,10 @@ export function registerSecurityUsageCollector({ usageCollection, config, licens
         };
       }
 
+      const legacyAuditLoggingEnabled = allowLegacyAuditLogging && config.audit.enabled;
       const auditLoggingEnabled =
-        (allowAuditLogging || allowLegacyAuditLogging) && config.audit.enabled;
+        allowAuditLogging && config.audit.enabled && config.audit.appender != null;
+
       const loginSelectorEnabled = config.authc.selector.enabled;
       const authProviderCount = config.authc.sortedProviders.length;
       const enabledAuthProviders = [
@@ -113,7 +115,7 @@ export function registerSecurityUsageCollector({ usageCollection, config, licens
       );
 
       return {
-        auditLoggingEnabled,
+        auditLoggingEnabled: legacyAuditLoggingEnabled || auditLoggingEnabled,
         loginSelectorEnabled,
         accessAgreementEnabled,
         authProviderCount,

```

Reply:
Good catch. We shouldn't really duplicate this logic (single source of truth) but I think it would be a fairly major undertaking to refactor this so will apply your fix for now.

Reply:
I agree on both counts!